src/csharp.tmLanguage.yml

```diff
@@ -3490,6 +3491,66 @@ repository:
       '4': { name: string.quoted.double.cs }
       '5': { name: string.quoted.double.cs }
 
+  preprocessor-app-directive:
+    begin: \s*(:)\s*
+    beginCaptures:
+      '1': { name: punctuation.separator.colon.cs }
+    end: (?=$)
+    patterns:
+      - include: '#preprocessor-app-directive-package'
```

Comment:
I'm torn between telling you to go add `sdk` support as well (and ask @jjonescz what other directives are supported), or just saying that we shouldn't do any of these and just do the generic version. Your call.